src/prefect/automations.py

```diff
@@ -0,0 +1,158 @@
+    EventTrigger,
+    MetricTrigger,
+    MetricTriggerOperator,
+    MetricTriggerQuery,
+    Posture,
+    PrefectMetric,
+    ResourceSpecification,
+    ResourceTrigger,
+    SequenceTrigger,
+    Trigger,
+)
+from prefect.exceptions import PrefectHTTPStatusError
+from prefect.utilities.asyncutils import sync_compatible
+
+__all__ = [
+    "AutomationCore",
+    "EventTrigger",
+    "ResourceTrigger",
+    "Posture",
+    "Trigger",
+    "ResourceSpecification",
+    "MetricTriggerOperator",
+    "MetricTrigger",
+    "PrefectMetric",
+    "CompositeTrigger",
+    "SequenceTrigger",
+    "CompoundTrigger",
+    "MetricTriggerQuery",
+]
+
+
+class Automation(AutomationCore):
+    id: Optional[UUID] = Field(default=None, description="The ID of this automation")
+
+    @classmethod
+    @sync_compatible
+    async def create(cls: Self, automation: Self) -> Self:
+        """
+        Create a new automation.
+
+        auto_to_create = Automation(
+            name="woodchonk",
+            trigger=EventTrigger(
+                expect={"animal.walked"},
+                match={
+                    "genus": "Marmota",
+                    "species": "monax",
+                },
+                posture="Reactive",
+                threshold=3,
+                within=timedelta(seconds=10),
+            ),
+            actions=[CancelFlowRun()]
+        )
+        created_automation = Automation.create(auto_to_create)
+        """
+        client, _ = get_or_create_client()
+        automation = AutomationCore(**automation.dict(exclude={"id"}))
+        automation_id = await client.create_automation(automation=automation)
+        automation = await client.read_automation(automation_id=automation_id)
+        automation = Automation(**automation.dict())
+        return automation if automation else None
+
+    @sync_compatible
+    async def update(self: Self):
+        """
+        Updates an existing automation.
+        auto = Automation.read(id = 123)
+        auto.name = "new name"
+        auto.update()
+        """
+
+        client, _ = get_or_create_client()
+        automation = AutomationCore(**self.dict(exclude={"id", "owner_resource"}))
+        await client.update_automation(automation_id=self.id, automation=automation)
+
+    @classmethod
+    @sync_compatible
+    async def read(
+        cls: Self, id: Optional[UUID] = None, name: Optional[str] = None
+    ) -> Self:
+        """
+        Read an automation by ID or name.
+        automation = Automation.read(name="woodchonk")
+
+        or
+
+        automation = Automation.read(id=UUID("b3514963-02b1-47a5-93d1-6eeb131041cb"))
+        """
+        if id and name:
+            raise ValueError("Only one of id or name can be provided")
+        if not id and not name:
+            raise ValueError("One of id or name must be provided")
+        client, _ = get_or_create_client()
+        if id:
+            automation = await client.read_automation(automation_id=id)
```

Comment:
If we could catch the `PrefectHTTPStatusError` here when reading a nonexistent id and maybe add a test for it, that might be nicer for the user than this error:
<details>
<pre><code>
In [3]: Automation.read(id='291ba287-2149-4379-825e-52b1d266f11')
---------------------------------------------------------------------------
PrefectHTTPStatusError                    Traceback (most recent call last)
Cell In[3], line 1
----> 1 Automation.read(id='291ba287-2149-4379-825e-52b1d266f11')

File ~/code-oss/prefect/src/prefect/utilities/asyncutils.py:296, in sync_compatible.<locals>.coroutine_wrapper(*args, **kwargs)
    294 # Run in a new event loop, but use a `Call` for nested context detection
    295 call = create_call(async_fn, *args, **kwargs)
--> 296 return call()

File ~/code-oss/prefect/src/prefect/_internal/concurrency/calls.py:432, in Call.__call__(self)
    430     return run_and_return_result()
    431 else:
--> 432     return self.result()

File ~/code-oss/prefect/src/prefect/_internal/concurrency/calls.py:318, in Call.result(self, timeout)
    312 def result(self, timeout: Optional[float] = None) -> T:
    313     """
    314     Wait for the result of the call.
    315 
    316     Not safe for use from asynchronous contexts.
    317     """
--> 318     return self.future.result(timeout=timeout)

File ~/code-oss/prefect/src/prefect/_internal/concurrency/calls.py:179, in Future.result(self, timeout)
    177     raise CancelledError()
    178 elif self._state == FINISHED:
--> 179     return self.__get_result()
    181 self._condition.wait(timeout)
    183 if self._state in [CANCELLED, CANCELLED_AND_NOTIFIED]:
    184     # Raise Prefect cancelled error instead of
    185     # `concurrent.futures._base.CancelledError`

File ~/.pyenv/versions/3.11.1/lib/python3.11/concurrent/futures/_base.py:401, in Future.__get_result(self)
    399 if self._exception:
    400     try:
--> 401         raise self._exception
    402     finally:
    403         # Break a reference cycle with the exception in self._exception
    404         self = None

File ~/code-oss/prefect/src/prefect/_internal/concurrency/calls.py:389, in Call._run_async(***failed resolving arguments***)
    387 with self.future.enforce_async_deadline() as cancel_scope:
    388     try:
--> 389         result = await coro
    390     finally:
    391         # Forget this call's arguments in order to free up any memory
    392         # that may be referenced by them; after a call has happened,
    393         # there's no need to keep a reference to them
    394         self.args = None

File ~/code-oss/prefect/src/prefect/automations.py:107, in Automation.read(cls, id, name)
    105 client, _ = get_or_create_client()
    106 if id:
--> 107     automation = await client.read_automation(automation_id=id)
    108     return Automation(**automation.dict()) if automation else None
    109 else:

File ~/code-oss/prefect/src/prefect/client/orchestration.py:3219, in PrefectClient.read_automation(self, automation_id)
   3216 if not self.server_type.supports_automations():
   3217     self._raise_for_unsupported_automations()
-> 3219 response = await self._client.get(f"/automations/{automation_id}")
   3220 if response.status_code == 404:
   3221     return None

File ~/code-oss/prefect/venvpydantic2/lib/python3.11/site-packages/httpx/_client.py:1801, in AsyncClient.get(self, url, params, headers, cookies, auth, follow_redirects, timeout, extensions)
   1784 async def get(
   1785     self,
   1786     url: URLTypes,
   (...)
   1794     extensions: RequestExtensions | None = None,
   1795 ) -> Response:
   1796     """
   1797     Send a `GET` request.
   1798 
   1799     **Parameters**: See `httpx.request`.
   1800     """
-> 1801     return await self.request(
   1802         "GET",
   1803         url,
   1804         params=params,
   1805         headers=headers,
   1806         cookies=cookies,
   1807         auth=auth,
   1808         follow_redirects=follow_redirects,
   1809         timeout=timeout,
   1810         extensions=extensions,
   1811     )

File ~/code-oss/prefect/venvpydantic2/lib/python3.11/site-packages/httpx/_client.py:1574, in AsyncClient.request(self, method, url, content, data, files, json, params, headers, cookies, auth, follow_redirects, timeout, extensions)
   1559     warnings.warn(message, DeprecationWarning)
   1561 request = self.build_request(
   1562     method=method,
   1563     url=url,
   (...)
   1572     extensions=extensions,
   1573 )
-> 1574 return await self.send(request, auth=auth, follow_redirects=follow_redirects)

File ~/code-oss/prefect/src/prefect/client/base.py:356, in PrefectHttpxClient.send(self, request, *args, **kwargs)
    353 response = PrefectResponse.from_httpx_response(response)
    355 if self.raise_on_all_errors:
--> 356     response.raise_for_status()
    358 return response

File ~/code-oss/prefect/src/prefect/client/base.py:169, in PrefectResponse.raise_for_status(self)
    167     return super().raise_for_status()
    168 except HTTPStatusError as exc:
--> 169     raise PrefectHTTPStatusError.from_httpx_error(exc) from exc.__cause__

PrefectHTTPStatusError: Client error '404 Not Found' for url 'https://api.prefect.cloud/api/accounts/43a83e5a-b23e-47cf-9652-5c3dc511f7e1/workspaces/75241259-81a4-4879-92aa-fc3f1df3e6e4/automations/291ba287-2149-4379-825e-52b1d266f11'
Response: {'detail': 'Not Found'}
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404
</code></pre>
</details>